pkg/syncer/metadatasyncer.go

```diff
@@ -369,20 +369,41 @@ func pvDeleted(obj interface{}, metadataSyncer *MetadataSyncInformer) {
 		return
 	}
 	var deleteDisk bool
-	if pv.Spec.ClaimRef != nil && (pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased) && pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
-		klog.V(3).Infof("PVDeleted: Volume deletion will be handled by Controller")
+	if pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
+		if pv.Spec.ClaimRef != nil {
+			if pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased {
+				var metadataList []cnstypes.BaseCnsEntityMetadata
+				pvMetadata := cnsvsphere.GetCnsKubernetesEntityMetaData(pv.Name, nil, true, string(cnstypes.CnsKubernetesEntityTypePV), pv.Namespace)
+				metadataList = append(metadataList, cnstypes.BaseCnsEntityMetadata(pvMetadata))
+
+				updateSpec := &cnstypes.CnsVolumeMetadataUpdateSpec{
+					VolumeId: cnstypes.CnsVolumeId{
+						Id: pv.Spec.CSI.VolumeHandle,
+					},
+					Metadata: cnstypes.CnsVolumeMetadata{
+						ContainerCluster: cnsvsphere.GetContainerCluster(metadataSyncer.cfg.Global.ClusterID, metadataSyncer.cfg.VirtualCenter[metadataSyncer.vcenter.Config.Host].User),
+						EntityMetadata:   metadataList,
+					},
+				}
+				klog.V(4).Infof("PVDeleted: vSphere provisioner removing volume metadata for the pv: %v. Volume deletion will be handled by Controller", pv)
+				if err := volumes.GetManager(metadataSyncer.vcenter).UpdateVolumeMetadata(updateSpec); err != nil {
+					klog.Errorf("PVUpdated: UpdateVolumeMetadata failed with err %v", err)
+				}
+			} else {
+				klog.V(4).Infof("PVDeleted: Volume %v is still in %v phase. Skipping metadata update.", pv.Name, pv.Status)
+			}
+		} else {
+			klog.V(4).Infof("PVDeleted: ClaimRef for the PV %v with Reclaim Policy %v is nil. Skipping metadata update", pv.Name, pv.Spec.PersistentVolumeReclaimPolicy)
+		}
+		klog.V(4).Infof("PVDeleted: Volume deletion will be handled by Controller")
```

Comment:
no need to print this message here. the controller will do its job when it receives a request to delete the volume.